txpipe/covariance.py

```diff
@@ -402,21 +402,44 @@ def compute_covariance(self, cosmo, meta, two_point_data):
         cl_cache = {}
         xi_pm = [[('plus','plus'), ('plus', 'minus')], [('minus','plus'), ('minus', 'minus')]]
 
+        print("N2pt", N2pt)
+        print("N2pt0", N2pt0)
+        
         # Look through the chunk of matrix, tracer pair by tracer pair
+        # Order of the covariance needs to be the cannonical order of saac. For a 3x2pt matrix that is:
+        # -galaxy_density_xi
+        # -galaxy_shearDensity_xi_t
+        # -galaxy_shear_xi_minus
+        # -galaxy_shear_xi_plus
+
+
         for i in range(N2pt):
             tracer_comb1 = tracer_combs[i]
 
-            if i == N2pt0:
+            # reset variable in the new loop
+            count_xi_pm2 = 0 
+
+            if i == N2pt0-(N2pt-N2pt0):
```

Comment:
This is a little confusing.  Can you phrase this as something like 
```
count_xi_pm1 = 1 if i in range(xim_start, xi_end) else 0
```
or similar, and the same for `count_xi_pm2` just so that it's more explicit?

Reply:
yes, thanks for the suggestion. I was so happy that it worked that didn't worry too much about the form!